trestle/core/markdown/docs_markdown_node.py

```diff
@@ -56,6 +56,14 @@ def __init__(self, key: str, content: DocsSectionContent, starting_line: int):
         super(DocsMarkdownNode, self).__init__(key, content, starting_line)
         self.content: DocsSectionContent = content
 
+    @classmethod
+    def build_tree_from_markdown(cls, lines: List[str], governed_header: Optional[str] = None):
+        """Construct a tree out of the given markdown."""
+        ob = cls.__new__(cls)
+        start_level = ob._get_max_header_lvl(lines)
```

Comment:
I see this `__new__` has been in the code for some time - but it is the only time it is used.  Just want to confirm there is a reason this classmethod is special and needs it.

Reply:
This is a fairly special use case for this class. This is a classmethod factory method to create the instance of MarkdownNode. The reason for this design choice was that each MarkdownNode instance operates on itself and builds the tree recursively. It needs to perform some subroutines (i.e. compute the max heading level, key, etc) and contain some information in each node (defined in __init__). Because we don't want to call `__init__` before we compute those this approach was used